src/ObjectMakr.ts

```diff
@@ -226,15 +225,17 @@ export class ObjectMakr implements IObjectMakr {
         // For each name in the current object:
         for (let name in base) {
             if (base.hasOwnProperty(name)) {
-                this.functions[name] = new Function() as IClassFunction;
+                if (!this.functions[name]) {
+                    this.functions[name] = new Function() as IClassFunction;
 
-                // This sets the Function as inheriting from the parent
-                this.functions[name].prototype = new parent();
-                this.functions[name].prototype.constructor = this.functions[name];
+                    // This sets the Function as inheriting from the parent
+                    this.functions[name].prototype = new parent();
+                    this.functions[name].prototype.constructor = this.functions[name];
+                }
 
                 // Add each property from properties to the Function prototype
                 for (let ref in this.properties[name]) {
-                    if (this.properties[name].hasOwnProperty(ref)) {
+                    if (this.properties[name].hasOwnProperty(ref) && !this.functions[name].prototype[ref]) {
```

Comment:
(this example is still in the realm of the Animal inheritance)
I was thinking that if an Animal constructor function was passed in but all of its properties (whatever those properties my be) weren't a part of it for some reason, they would be added here `this.functions[name].prototype[ref] = this.properties[name][ref];`
i.e. letting it go through 
```javascript
this.functions[name] = function(){};

// This sets the Function as inheriting from the parent
this.functions[name].prototype = new parent();
this.functions[name].prototype.constructor = this.functions[name]; 
```
and this section
```javascript
 for (let ref in this.properties[name]) {
    if (this.properties[name].hasOwnProperty(ref) && !this.functions[name].prototype[ref]) {
        this.functions[name].prototype[ref] = this.properties[name][ref];
    }
} 
```
wouldn't change the end result of the function

Unless you were implying that a function that was passed into the objectMakr constructor should not be changed by `processFunctions`
Does that make more sense or no?